src/info-panel/InfoPanel.js

```diff
@@ -0,0 +1,85 @@
+import useBreakpoint from "@w11r/use-breakpoint";
+import PropTypes from "prop-types";
+import React from "react";
+import styled from "styled-components";
+import { ReactComponent as MenuOpenIcon } from "../assets/icons/menuOpen.svg";
+import { useInfoPanelDispatch } from "./InfoPanelContext";
+
+const InfoPanelWrapper = styled.div`
+  background: ${(props) => props.theme.colors.tooltipBackground};
+  border-radius: 0;
+  bottom: 0;
+  left: 0;
+  padding-top: 8px;
+  position: fixed;
+  right: 0;
+  z-index: ${(props) => props.theme.zIndex.modal};
+`;
+
+const InfoPanelOverlay = styled.div`
+  background: ${(props) => props.theme.colors.tooltipBackground};
+  bottom: 0;
+  left: 0%;
+  opacity: 0.1;
+  position: fixed;
+  right: 0;
+  top: 0;
+  z-index: ${(props) => props.theme.zIndex.modal};
+`;
+
+const CloseButtonWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const CloseButton = styled.button`
+  background: none;
+  border: none;
+  cursor: pointer;
+  padding: 8px;
+  margin-right: 24px;
+
+  rect {
+    fill: ${(props) => props.theme.colors.bodyLight} !important;
+  }
+`;
+
+const ICON_SIZE = 16;
+
+export default function InfoPanel({ data, renderContents }) {
+  const enabled = useBreakpoint(false, ["mobile-", true]);
+  const infoPanelDispatch = useInfoPanelDispatch();
+
+  const dismiss = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    infoPanelDispatch({ type: "clear" });
+  };
+
+  if (enabled && data && renderContents) {
+    return (
+      <>
+        <InfoPanelOverlay onClick={dismiss} />
+        <InfoPanelWrapper className="InfoPanel">
+          <CloseButtonWrapper>
+            <CloseButton onClick={dismiss}>
+              <MenuOpenIcon width={ICON_SIZE} height={ICON_SIZE} />
+            </CloseButton>
+          </CloseButtonWrapper>
+          {renderContents(data)}
```

Comment:
I'm probably missing something, but if we're always calling `renderContents` with the `data` that is being passed in as props, should we instead just change this to a single prop called `content` and leave the responsibility for joining the data and rendering functionality up to the calling component?

Reply:
yeah I guess the issue that I ran into here is that I wanted there to be only one of these, unlike tooltips that are created and destroyed on command. that necessitated this somewhat convoluted setup of passing data and a "render prop" to it via a Context. Because in effect the content is not actually controlled by the parent component.  (this addresses your other question too — I'm not so sure it's a good idea, or even really possible, to store the _output_ of a render function in a context? I can't say exactly why but it just feels .... wrong somehow?)

Now that I am at the end of this looking back though, I'm not sure that all this was really necessary. It does feel a bit convoluted ... will take another look and see if I can simplify (and hopefully address that bug you saw in the process)

Reply:
ok so I dug into this a little bit and it seems like the answer to "can I simplify this" is .......... no

at least, not easily. the way Semiotic renders tooltips is pretty tightly coupled to a) hover events and b) its "annotations" feature. there's not really a straightforward way I can find to extricate an element from that structure and have it take over the whole viewport as we want the InfoPanel to do. 

there may be an alternative where each `ResponsiveTooltipController` renders its own `InfoPanel`s on demand instead of communicating with one centralized one; I'm not sure whether that would be better than this, honestly. And this is already done!